testing/integration/proxy_url_test.go

```diff
@@ -392,19 +392,29 @@ func TestProxyURL_RemoveProxyFromThePolicy(t *testing.T) {
 	p.assertConnectedFleet(t)
 }
 
+// assertConnectedFleet is deprecated, use AssertConnectedToFleet instead
+// Deprecated.
 func (p *ProxyURL) assertConnectedFleet(t *testing.T) {
+	AssertConnectedToFleet(t, p.fixture)
+}
+
+func AssertConnectedToFleet(t *testing.T, f *integrationtest.Fixture) {
 	t.Helper()
 
 	var err error
 	var agentStatus integrationtest.AgentStatusOutput
-	if !assert.Eventually(t, func() bool {
-		agentStatus, err = p.fixture.ExecStatus(context.Background())
+	assertFn := func() bool {
+		agentStatus, err = f.ExecStatus(context.Background())
 		return agentStatus.FleetState == int(cproto.State_HEALTHY)
-	}, 5*time.Minute, 5*time.Second,
+	}
+
+	connected := assert.Eventually(t, assertFn, 5*time.Minute, 5*time.Second,
 		"want fleet state %s, got %s. agent status: %v",
-		cproto.State_HEALTHY, cproto.State(agentStatus.FleetState), agentStatus) {
+		cproto.State_HEALTHY, cproto.State(agentStatus.FleetState), agentStatus)
+
+	if !connected {
 		if err != nil {
```

Comment:
Nit: could be simplified.
```suggestion
	if !connected && err != nil {
```